src/nft_contract/src/main.nr

```diff
@@ -50,118 +47,120 @@ pub contract NFT {
         minter: PublicImmutable<AztecAddress, Context>,
     }
 
+    /// @dev Initializes the token with a minter
+    /// @param name The name of the NFT
+    /// @param symbol The symbol of the NFT
+    /// @param minter The address of the minter
     #[public]
     #[initializer]
     fn constructor_with_minter(name: str<31>, symbol: str<31>, minter: AztecAddress) {
         storage.name.initialize(FieldCompressedString::from_string(name));
         storage.symbol.initialize(FieldCompressedString::from_string(symbol));
-
         storage.minter.initialize(minter);
     }
 
     /** ==========================================================
      * ========================= PRIVATE =========================
      * ======================================================== */
 
-    // Transfers token with `token_id` from public balance of message sender to a private balance of `to`.
     #[private]
-    fn transfer_to_private(to: AztecAddress, token_id: Field) {
-        let from = context.msg_sender();
-
-        let nft = NFT::at(context.this_address());
-
-        // We prepare the private balance increase.
-        let commitment = _prepare_private_balance_increase(to, &mut context, storage);
-
-        // At last we finalize the transfer. Usage of the `unsafe` method here is safe because we set the `from`
-        // function argument to a message sender, guaranteeing that he can transfer only his own NFTs.
-        nft._finalize_transfer_to_private_unsafe(from, token_id, commitment).enqueue(&mut context);
+    fn transfer_private_to_public(from: AztecAddress, to: AztecAddress, token_id: Field, nonce: Field) {
+        _validate_from_private(&mut context, from, nonce);
```

Comment:
not important at all but let's add a space between authwit calls and contract logic calls so it's easier to differentiate while reading